### PR TITLE
Add missing RPC parts

### PR DESCRIPTION
**Pull request by noise64**
Resolves: https://github.com/golemcloud/docs/issues/53
Resolves: https://github.com/golemcloud/docs/issues/60
Resolves: https://github.com/golemcloud/docs/issues/76

### CODE REVIEW

**Review by justcoon — APPROVED** [review]

File: src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx [hunk 1]
```diff
@@ -1,6 +1,266 @@
...
+# the above will execute: jco stubgen wit -o src/generated
+```
+
+### Creating the worker client
+
+The worker to be invoked is identified by an URI which consists of the **component ID** and the **worker name**.
+
+If the worker pointed by the URI does not exists, it is going to be created automatically and it inherits the environment variables of the caller.
+
+A common pattern is to use environment variables to configure the deployed component's **component ID** and use that to construct the URI. See the [defining components](/docs/go-language-guide/defining-components) page for more information about passing configuration values to workers.
```
Comment (justcoon) [inline src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx:183]: `/docs/go-language-guide/defining-components` is this proper link for TS ?

File: src/pages/docs/experimental-languages/js-language-guide/rpc.mdx [hunk 2]
```diff
@@ -1,6 +1,242 @@
...
 import { Steps } from "nextra/components"
 
-# Worker to Worker communication for Javascipt components
+# Worker to Worker communication for JavaScript components
 
 See the [Worker to Worker communication page](/docs/concepts/worker-to-worker-communication) for a general overview of how workers can invoke each other in Golem.
+
+## Tooling setup
+
+Make sure that all required tools are installed according to the [setup](/docs/experimental-languags/kss-language-guide/setup) page, including the _Worker to Worker communication_ parts.
```
Comment (justcoon) [inline src/pages/docs/experimental-languages/js-language-guide/rpc.mdx:10]: `(/docs/experimental-languags/kss-language-guide/setup)` look like typos in link

File: src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx (hunk 1 above)
Comment (noise64) [reply src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx]: https://github.com/golemcloud/docs/pull/78/commits/b02500d6f82a48e816d16e163243b2b9a1cfd92f

File: src/pages/docs/experimental-languages/js-language-guide/rpc.mdx (hunk 2 above)
Comment (noise64) [reply src/pages/docs/experimental-languages/js-language-guide/rpc.mdx:10]: https://github.com/golemcloud/docs/pull/78/commits/b02500d6f82a48e816d16e163243b2b9a1cfd92f

File: src/pages/docs/experimental-languages/js-language-guide/rpc.mdx [hunk 3]
```diff
@@ -1,6 +1,242 @@
...
+}
+```
+
+### Creating the worker client
+
+The worker to be invoked is identified by an URI which consists of the **component ID** and the **worker name**.
+
+If the worker pointed by the URI does not exists, it is going to be created automatically and it inherits the environment variables of the caller.
+
+A common pattern is to use environment variables to configure the deployed component's **component ID** and use that to construct the URI. See the [defining components](/docs/go-language-guide/defining-components) page for more information about passing configuration values to workers.
```
[the remark this answers is not in the document]
Comment (noise64) [reply src/pages/docs/experimental-languages/js-language-guide/rpc.mdx]: https://github.com/golemcloud/docs/pull/78/commits/33c79693cc361d76b538f13ab0e6e4ed1521391d

**Review by justcoon — APPROVED** [review]

File: src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx [hunk 4]
```diff
@@ -1,6 +1,266 @@
...
+Let's create one for component2 using `golem-cli stubgen build`:
+
+```shell
+$ golem-cli stubgen build --source-wit-root component2/wit --dest-wasm component2-stub/component2_stub.wasm --dest-wit-root component2-stub/wit
+```
+
+The above command will place the generated **WIT** definitions and **WASM** component into the `component2-stub` folder:
+
+```shell
+$ tree component2-stub
```
[the remark this answers is not in the document]
Comment (nicoburniske) [reply src/pages/docs/experimental-languages/ts-language-guide/rpc.mdx:107]: https://nextra.site/docs/guide/built-ins/filetree

**Review by nicoburniske — COMMENTED** [review]
I would swap the `tree` shells for the FileTree component

**Pull request merged**